components/notion/notion.app.mjs

```diff
@@ -29,6 +29,42 @@ export default {
         return this._buildPaginatedOptions(options, response.next_cursor);
       },
     },
+    propertyId: {
+      type: "string[]",
+      label: "Property ID",
+      description: "The identifier for a Notion page property",
+      optional: true,
+      async options({ pageId }) {
+        const response = await this.retrievePage(pageId);
+
+        const parentType = response.parent.type;
+        try {
+          const { properties }  =
+          parentType === "database_id"
+            ? await this.retrieveDatabase(response.parent.database_id)
+            : response;
+
+          const propEntries = Object.entries(properties);
+          const propIds  = propEntries.length === 1 && Object.values(propEntries)[0][1].id === "title"
+            ?
+            propEntries.map((prop) => {
+              return {
+                label: prop[1].type,
+                value: prop[1].id,
+              };
+            })
+            : propEntries.map((prop) => {
+              return {
+                label: prop[1].name,
+                value: prop[1].id,
+              };});
```

Comment:
**Minor/Optional**
Use parenthesis instead of `return` statement

```suggestion
            : propEntries.map((prop) => ({
                label: prop[1].name,
                value: prop[1].id,
              }));
```